packages/vite/src/node/plugins/resolve.ts

```diff
@@ -474,11 +474,14 @@ function tryFsResolve(
 ): string | undefined {
   const { file, postfix } = splitFileAndPostfix(fsPath)
 
+  // Dependencies like es5-ext use `#` in their paths. We don't support `#` in user
+  // source code so we only need to perform the check for dependencies.
+  const trySharp = fsPath.includes('#') && fsPath.includes('node_modules')
+
   let res: string | undefined
 
-  // if there is a postfix, try resolving it as a complete path first (#4703)
   if (
```

Comment:
I think we can add some comments before each `if` branch. The comments  tell what `fsPath` will hit the `if` condition. It's helpful to understand these codes quickly.

Reply:
I renamed the variable instead of adding a new comment because I think the main comment for `tryUnsplitted` is enough now.